visivo/models/model.py

```diff
@@ -1,12 +1,53 @@
-from typing import Optional
-from pydantic import Field, validator
+from typing import Any, List, Optional, Union
+from typing_extensions import Annotated
+from pydantic import Field, Discriminator, Tag
 from visivo.models.base.named_model import NamedModel
 from visivo.models.base.parent_model import ParentModel
-from visivo.models.target import DefaultTarget, Target
-from .base.base_model import generate_ref_field
+from visivo.models.target import DefaultTarget, Target, TypeEnum
+from .base.base_model import RefString, generate_ref_field
+import pandas
+import re
 
 
-class Model(NamedModel, ParentModel):
+class Model(NamedModel):
+    pass
+
+
+class RunModel(Model):
+    cmds: List[str] = Field(description="The sql used to generate your base data")
+
+    @property
+    def sql(self):
+        return f"select * from {self.table_name}"
+
+    @property
+    def table_name(self):
+        return re.sub(r"[^a-zA-Z0-9_]", "_", self.name)
+
+    def get_target(self, output_dir):
+        return Target(
+            name=f"model_{self.name}_generated_target",
+            type=TypeEnum.sqlite,
+            database=self.get_database(output_dir),
+        )
+
+    def get_database(self, output_dir):
+        return f"{output_dir}/{self.table_name}.sqlite"
+
+    def insert_csv_to_sqlite(self, output_dir):
+        from sqlalchemy import create_engine
+        import subprocess
+
+        csv_file = f"{output_dir}/{self.name}.csv"
+        with open(csv_file, "w+") as file:
+            subprocess.run(self.cmds, stdout=file, stderr=subprocess.STDOUT, text=True)
+
+        engine = create_engine(f"sqlite:///{self.get_database(output_dir)}")
+        data_frame = pandas.read_csv(csv_file)
+        data_frame.to_sql(self.name, engine, if_exists="replace", index=True)
```

Comment:
We could eventually support incremental updates pretty easily by passing through a param that allows them to append

Reply:
Correct.